crates/router/src/core/payments/tokenization.rs

```diff
@@ -199,22 +199,64 @@ where
                     .await?;
                 let customer_id = customer_id.to_owned().get_required_value("customer_id")?;
                 let merchant_id = merchant_account.get_id();
-                let (mut resp, duplication_check) = if !state.conf.locker.locker_enabled {
-                    skip_saving_card_in_locker(
-                        merchant_account,
-                        payment_method_create_request.to_owned(),
-                    )
-                    .await?
-                } else {
-                    pm_status = Some(common_enums::PaymentMethodStatus::from(
-                        save_payment_method_data.attempt_status,
-                    ));
-                    Box::pin(save_in_locker(
-                        state,
-                        merchant_account,
-                        payment_method_create_request.to_owned(),
-                    ))
-                    .await?
+                let is_network_tokenization_enabled =
+                    merchant_account.is_network_tokenization_enabled;
+                let ((mut resp, duplication_check, network_token_requestor_ref_id), token_resp) =
+                    if !state.conf.locker.locker_enabled {
+                        let (res, dc) = skip_saving_card_in_locker(
+                            merchant_account,
+                            payment_method_create_request.to_owned(),
+                        )
+                        .await?;
+                        ((res, dc, None), None)
+                    } else {
+                        pm_status = Some(common_enums::PaymentMethodStatus::from(
+                            save_payment_method_data.attempt_status,
+                        ));
+                        let (res, dc, _ref_id) = Box::pin(save_in_locker(
+                            state,
+                            merchant_account,
+                            Some(&save_payment_method_data.request.get_payment_method_data()),
+                            payment_method_create_request.to_owned(),
+                            false, //saving the card in locker
+                            amount,
+                            currency,
+                        ))
+                        .await?;
+
+                        if is_network_tokenization_enabled {
```

Comment:
Can we move this check entirely out of the tuple and just store the card in the locker first and later on  if this flag is enabled we can also save the token details in the locker, given so we can make storing the network_token in locker and card in locker as independent functions